chandra_aca/centroid_resid.py

```diff
@@ -0,0 +1,244 @@
+                q_mis_inv = np.repeat(Quat(acal['aca_misalign'][0]).inv().q,
+                                      len(asol)).reshape((4, len(asol))).transpose()
+                # Quaternion multiply the asol quats with that inv misalign and save
+                # I could also do this with the transform matrix and then only need
+                # one accessory quat function.
+                att_chunks.append(quat_vmult(asol['q_att'], q_mis_inv))
+                time_chunks.append(np.array(asol['time']))
+            atts = np.vstack(att_chunks)
+            att_times = np.hstack(time_chunks)
+        self.atts = atts  # (N, 4) numpy array
+        self.att_times = att_times
+
+    def set_star(self, agasc_id=None, obsid=None, slot=None, date=None):
+        """
+        Set self.ra and dec from either agasc_id *or* obsid + slot.
+
+        This assumes use of star in default agasc miniagasc (no 1.4 or 1.5 or very faint stars)
+
+        Along the way set self.agasc_id.  One can also just set
+        self.ra and dec directly.
+        """
+        if obsid is not None and slot is not None:
+            if date is None:
+                raise ValueError("Need date to look up star catalog if obsid/slot provided")
+            sc = mica.starcheck.get_starcheck_catalog_at_date(date)
+            star = sc['cat'][(sc['cat']['slot'] == slot)
+                             & ((sc['cat']['type'] == 'GUI') | (sc['cat']['type'] == 'BOT'))]
+            if not len(star):
+                raise ValueError(
+                    "No GUI or BOT in slot {} for obsid {} in dwell".format(slot, obsid))
+            agasc_id = star['id'][0]
+        # Could also add logic to infer star from loose position and magnitude
+        star = agasc.get_star(agasc_id, date=date)
+        self.agasc_id = agasc_id
+        self.ra = star['RA_PMCORR']
+        self.dec = star['DEC_PMCORR']
+
+    @property
+    def yags(self):
+        return self._yags
+
+    @yags.setter
+    def yags(self, vals):
+        if isinstance(vals, fetch.MSID):
+            self._yags = np.array(vals.vals)
+            self._yag_times = vals.times
+        else:
+            self._yags = np.array(vals)
+
+    @property
+    def yag_times(self):
+        return self._yag_times
+
+    @yag_times.setter
+    def yag_times(self, vals):
+        self._yag_times = np.array(vals)
+
+    @property
+    def zags(self):
+        return self._zags
+
+    @zags.setter
+    def zags(self, vals):
+        if isinstance(vals, fetch.MSID):
+            self._zags = np.array(vals.vals)
+            self._zag_times = vals.times
+        else:
+            self._zags = np.array(vals)
+
+    @property
+    def zag_times(self):
+        return self._zag_times
+
+    @zag_times.setter
+    def zag_times(self, vals):
+        self._zag_times = np.array(vals)
+
+    def get_residuals(self):
+        """
+        Return dyags and dzags, corresponding to the centroid residuals
```

Comment:
Confused a bit that it looks like you comments were outdated before they got to github.